### PR TITLE
tests(resharding): add a resharding test with single-shard tracking and shard shuffling

### PR DESCRIPTION
**Pull request by marcelo-gonzalez**
When testing resharding and state sync integration, the existing testloop tests with some modifications are very helpful. So this makes a few edits to the basic resharding tests and adds a test that has nodes tracking only one shard and enables shard shuffling. For now we `#[ignore]` it because it fails, but we can enable it once everything is fixed.

### CODE REVIEW

**Comment by marcelo-gonzalez** [comment]
right now the failure happens in `Chain::get_state_sync_info()` but if you comment out the assertion there that says the shard layout should not change, you push the failure further out to the current actually broken thing. For now just going to leave that assertion in place but we can remove it later

File: integration-tests/src/test_loop/tests/resharding_v3.rs [hunk 1]
```diff
@@ -169,6 +199,21 @@ impl TestReshardingParameters {
...
+
+    fn track_all_shards(mut self, track_all_shards: bool) -> Self {
+        self.track_all_shards = track_all_shards;
+        self
+    }
+
+    fn all_chunks_expected(mut self, all_chunks_expected: bool) -> Self {
+        self.all_chunks_expected = all_chunks_expected;
+        self
+    }
```
Comment (Longarithm) [inline integration-tests/src/test_loop/tests/resharding_v3.rs:216]: Yeah that's why I suggested @Trisfald to have a Builder pattern here...

Anyway, that's probably fine as is, let's just drop bool arguments and call method only if we need to change the default: `shuffle_shard_assignment`, `single_shard_tracking`, `dont_assert_empty_chunks (?!)`. It's not like we want to call these methods twice, ever.

**Review by Longarithm — APPROVED** [review]

File: integration-tests/src/test_loop/tests/resharding_v3.rs [hunk 2]
```diff
@@ -169,6 +199,21 @@ impl TestReshardingParameters {
...
+        self.shuffle_shard_assignment_for_chunk_producers = true;
+        self
+    }
+
+    fn single_shard_tracking(mut self) -> Self {
+        self.track_all_shards = false;
+        self
+    }
+
+    fn chunk_miss_expected(mut self) -> Self {
```
Comment (Longarithm) [inline integration-tests/src/test_loop/tests/resharding_v3.rs:213]:
```suggestion
    fn chunk_miss_possible(mut self) -> Self {
```

etc

Comment (marcelo-gonzalez) [reply integration-tests/src/test_loop/tests/resharding_v3.rs]: done

**Pull request merged**